cgi-bin/DW/Setting/StickyEntry.pm

```diff
@@ -48,11 +70,38 @@ sub option {
 sub save {
     my ( $class, $u, $args ) = @_;
 
-    my $sticky = $class->get_arg( $args, "stickyid" ) || '';
-    $sticky = LJ::text_trim( $sticky, 0, 100 );
-    unless ( $u->sticky_entry ( $sticky ) ) {
-        $class->errors( "stickyid" => $class->ml( 'setting.stickyentry.error.invalid' ) ) ;
+    my $max_sticky_count = $u->count_max_stickies;
+    my @stickies;
+    # Create a hash that we will use to check for duplicate entries.
+    my %unique = ();
+    my $username = $u->user;
+
+    for my $i ( 1 ... $max_sticky_count ) {
+        my $stickyi = $class->get_arg( $args, "stickyid${i}" ) || '';
+        # Unless this is a blank form entry...
+        unless ( $stickyi eq '' ) {
```

Comment:
Hmmm I think "next unless $stickyi" because the original, I just realized, is a double-negative.